building-and-packaging/material/systempaths-and-librarytools_slides.md

```diff
@@ -0,0 +1,289 @@
+      color: orange;
+  }
+  .reveal section img {
+    background:none;
+    border:none;
+    box-shadow:none;
+  }
+</style>
+
+
+# System Paths, Libraries, and How to Use Them
+
+---
+
+## Introduction
+
+- Common functionality should not be redone the whole time
+    - Examples: [C++ Standard Library](https://en.cppreference.com/w/cpp), [Python Standard Library](https://docs.python.org/3/library/), [PETSc](https://petsc.org/), [FEniCS](https://fenicsproject.org/)...
+- For reusing software we need to know
+    - where the software is located
+    - how to compile and link with/against
+
+---
+
+## Static and shared libraries
+
+- Static
+    - Libraries: `NAME.a`
+    - Self-contained code/libraries
+    - Larger library/executable
+- Shared
+    - Libraries: `NAME.so`
+    - Shared code must be installed and accessible
+    - Executables can share library in memory
+
+---
+
+## Filesystem Hierarchy Standard
+
+- Defines filesystem layout and location of common files on Linux
+
+<img src="https://raw.githubusercontent.com/Simulation-Software-Engineering/Lecture-Material/add-overview-systempaths-librarytools/building-and-packaging/material/figs/filesystem_hierarchy_structure/fig.png" width=100%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+- [Official homepage of FHS](https://refspecs.linuxfoundation.org/fhs.shtml)
+- Certain structure should
+
+**TODO** Set this link to the GitHub main branch before merging.
+
+---
+
+## Some Environment Variables
+
+- `HOME` or `~`: User's home directory
+- `PWD`: Current working directory
+- `PATH`: Location of executable
+- `LC_<NAME>`: Locale's
+- `LANG`: Systems' language
+- System variables often set from `/etc/profile` and `/etc/bashrc` or `/etc/bash.bashrc`
+
+---
+
+## Manipulating Environment Variables 1/2
+
+- Show content of paths via
+
+  ```bash
+  echo $VARIABLENAME
+  printenv VARIABLENAME
+  ```
+
+- Setting variable for a single command (bash)
+
+  ```bash
+  ENVVARIABLE=VALUE ./COMMAND
+  ```
+
+- Setting **environment** variable
+
+  ```bash
+  export ENVVARIABLE=VALUE
+  ```
+
+---
+
+## Manipulating Environment Variables 2/2
+
+- Adding additional values by colon (`:`) separated list
+
+  ```bash
+  PATH=ADDITONALPATH:${PATH}
+  ```
+
+  ```bash
+  PATH=/home/jaustar/bin:${PATH}
+  ```
+
+- Extra paths of user often set in `${HOME}/.bashrc`, `${HOME}/.bash_exports`, `${HOME}/.bash_profile`,...
+
+---
+
+## Some Environment Variables (C++)
+
+- `LD_<NAME>`: Loader related variable
+    - `LD_LIBRARY_PATH`: Library paths for runtime
+- `LIBRARY_PATH`: Library path for compilation (linker)
+- `CPATH`, `C_INCLUDE_PATH`, `CPLUS_INCLUDE_PATH`: Include file path
+- Names can vary beween OS, software.
+- Variables should be manipulated carefully.
+
+---
+
+## Paths and Environment Variables (Python)
+
+- `/usr/`: System-wide installation
+    - `/usr/bin`: Executable scripts
+    - `/usr/lib/pythonX.Y`: Modules/packages
+- `${HOME}/.local`: As above
```

Comment:
```suggestion
- `${HOME}/.local`: User installation (`--user`)
```
?
Or what do you mean by "as above"?

Reply:
I meant that is the same "base" level as the `/usr/` directory and follows a similar structure on lower level. I will rephrase this and/or replace it with a figure.

Reply:
I added a figure now that should make this slide easier to read. There might still be an old figure cached by GitHub when one tries to inspect the slides.